data_collection/web/guides.rst

```diff
@@ -40,8 +40,47 @@ Example::
 
 .. todo:: Explain how to change page URL and title for virtual page views or link to SPA instructions page.
 
+User ID
+-------
+
+User ID is an additional session dimension in yours analytics data. When set you will be able to aggregate data by this dimension. You can use Tag Manger, default Piwik Pro tag to setup `User ID` without editing any JS code. To set it manually follow this guide.
+
+To set up `User ID` dimension add a call to :ref:`setUserId<jtc-api-setUserId>` to your tracking code:
+
+.. code-block:: javascript
+
+    _paq.push(["setUserId", "user-name@example.com"]);
+
+.. note ::
+
+    #. `User ID` can't be longer than 1024 bytes, it will be 1024 characters unless you use only ASCII characters. When using Unicode check the length of user identifier, so you won't exceed 1024 bytes.
```

Comment:
I think this information should not be in a guide but in lower level -> function description.

Reply:
It is requirement of our CPP, this is special dimension and it is better to warn user early than later looking up for issues "why I cant find users by their ID". It is an edge case, but our customers surprised us more than once.

Reply:
I'd suggest to add important info: invoking `setUserId` won't send any info to the tracker on its own. It will be sent with next subsequent event. Do we want to include info that it can be submitted with ping?

Reply:
- Added info that you have send `User ID` with additional request
- Left info about `User ID` length

Reply:
```suggestion
    #. `User ID` can't be longer than 1024 bytes. It will be 1024 characters if you use only ASCII characters, but Unicode characters may require more bytes per character, so you should make sure that Unicode identifier cut down to 1024 bytes is still unique.
```